### ISSUE TITLE
Only one event on remote is send

### ISSUE THREAD

**Issue by sschnake** [Leggin/dirigera#28]
When subscribing to the events I do only get one event on pressing the remote 
`firmware_version='1.0.47', hardware_version='1', model='RODRET Dimmer', manufacturer='IKEA of Sweden',`
Need to reboot the hub to get the event only one time again.
I do only have remotes on the hub as I wanted to use the Ikea devices to control other home devices. Current plan is to get all the remotes to mqtt.

**Comment by Leggin** [comment]
This sounds strange, but from the description I can't really think of a fix. Can you share your code?

**Comment by bwiklak** [comment]
Hi, I also have question about remotes.
Is there a way to listen to events on remote in order to create own automations?
I can list my controllers using `get_controllers` but I'm not receiving any messages from it.
I have it currently linked to a bulb and I only see events from the bulb:

    def on_message(ws: Any, message: str):
        message_dict = json.loads(message)
        data = message_dict["data"]
        print(data)
    
    def on_error(ws: Any, message: str):
        print(message)
    
    hub.create_event_listener(
        on_message=on_message, on_error=on_error
    )

Example:

`{'id': 'a83fc40d-3e29-4b63-b6c2-f61e5ccbbffa_1', 'type': 'light', 'deviceType': 'light', 'createdAt': '2024-02-20T20:19:10.000Z', 'isReachable': True, 'lastSeen': '2024-02-21T02:02:21.000Z', 'attributes': {'isOn': True}, 'remoteLinks': ['ce33f4cc-904b-4520-9c38-135352b576c6_1', 'e2e1d184-ad4f-4b53-ba05-f47cda3fd8b8_1']}`

`ce33f3cc` is my Styrbar controller I'm currently testing.

**Comment by TomyMMX** [comment]
@bwiklak I actually think the remotes don't publish their events, sadly.
My workaround is to create "fake" scenes that don't do anything (in the IKEA app) and then tie them to the controllers (at least that works for the action button).
Then I just wait until that scene gets triggered and check if the trigger was my button.
It is a bit annoying, but it works well enough.

**Comment by bwiklak** [comment]
Right, thanks for answer, I already bought action buttons so this should work.

**Comment by Leggin** [comment]
@TomyMMX that's a cool work around!
With the normal remote control it does not seem to work sadly so I can't test it.
But I thinks this would be a great addition to the readme for others to use in this scenario, would you be interested in opening a PR with these infos?

**Comment by slajob** [comment]
> @bwiklak I actually think the remotes don't publish their events, sadly. My workaround is to create "fake" scenes that don't do anything (in the IKEA app) and then tie them to the controllers (at least that works for the action button). Then I just wait until that scene gets triggered and check if the trigger was my button. It is a bit annoying, but it works well enough.

 How you can create a fake scenes? I can't create empty scene as provided in screenshot: 
![IMG_948D7646D9B3-1](https://github.com/Leggin/dirigera/assets/33373409/6d631e6e-ed52-4dc5-829f-cc5f4c46b07f)
What's your workaround? 

@Leggin How you are exploring dirigiera api? I didn't see anything about single, double and long press event :/ It will be perfect to get this actions :P

**Comment by TomyMMX** [comment]
@slajob yeah, scenes can't be empty.
I have a light bulb that I only use when I go to sleep in the evening, so that "fake" scene just turns that off. Which is fine as it is always off when I use my action button.
Not ideal, I know, but works for me.

**Comment by slajob** [comment]
Okey, maybe burned out old tradfri light could be perfect for that

**Comment by Leggin** [comment]
@slajob the first version for scene manipulation is release, please let me know if you encounter any problems.

**Comment by slajob** [comment]
> @slajob the first version for scene manipulation is release, please let me know if you encounter any problems.

Thanks! I'll try it and give you some feedback

**Comment by ValonVN0** [comment]
Apologies, slight necro post,

Testing this at the moment with both the [STYRBAR](https://www.ikea.com/gb/en/p/styrbar-remote-control-smart-white-60488366/) and [RODRET](https://www.ikea.com/gb/en/p/rodret-wireless-dimmer-power-switch-smart-white-80559796/) and unfortunately it only seems to register any actions via one of the buttons.

What I've attempted is looping through the controllers, then an int loop of say, 5 or so, and attempting to create a new scene using this module using that int as the button index, and referencing the int in the scene name too (I used the remote device id, followed by _ and the int, my hopes being I could then check for that ID when the hub event fires)

As mentioned, unfortunately it only connects to the one button on each remote.

I'd link it to a dummy plug for example using the app itself however it's incredibly limited, it'll either just be on or off.
